ggshield/core/config/utils.py

```diff
@@ -80,61 +65,42 @@ def get_global_path(filename: str) -> str:
     return os.path.join(os.path.expanduser("~"), filename)
 
 
-class YAMLFileConfig:
-    """Helper class to define configuration object loaded from a YAML file"""
-
-    def __init__(self, **kwargs: Any) -> None:
-        raise NotImplementedError
-
-    def to_dict(self) -> Union[List, Dict]:
-        return custom_asdict(self, root=True)
+class ConfigBaseModel(BaseModel):
+    class Config:
+        extra = Extra.forbid
 
-    def update_config(self, data: Dict[str, Any]) -> bool:
+    def update_from_model(self, other: BaseModel) -> None:
         """
-        Update the current config, ignoring the unrecognized keys
+        Update `self` with fields from `other` if they are set.
+        `self` and `other` are expected to be the same type.
         """
-        field_names = {field_.name for field_ in fields(self)}
-        for key, item in data.items():
-            if key not in field_names:
-                click.echo("Unrecognized key in config: {}".format(key))
-                continue
-            if isinstance(getattr(self, key), list):
-                getattr(self, key).extend(item)
-            elif isinstance(getattr(self, key), set):
-                getattr(self, key).update(item)
+        assert self.__class__ is other.__class__
```

Comment:
For now we don't have subconfigs but could it make sense to check if the `other` class is a parent of the `self` class?